### DISCUSSION TITLE
Unclear Documentation on the Net power consumption setting

### DISCUSSION THREAD

**Discussion by splitbrain** [jmcollin78/solar_optimizer#51] [Q&A] [answered]
I'm not a 100% sure what I should configure for the net power consumption. I have entities for the following:

* current **import** from grid
* current **export** to grid
* current power **consumption** in my home (uses electricity from battery, solar and grid as necessary)

Of course I can combine and modify the values from these sensors with a helper. But I am not a 100% sure what is needed here.

From the description of the field it might be  `import * -1` or maybe it's `(import * -1) + export`? Could you clarify this in the docs?

**Comment by jmcollin78** [comment, marked as answer]
Hello @splitbrain ,

As explain in the readme, net consumption is: "the sensor which gives the instantaneous net consumption of the dwelling (it must be negative if production exceeds consumption). This figure is indicated in Watt,"

So it import - export in your case: it is positive if your import is greater that export and negative on the contrary.

**Comment by jmcollin78** [comment]
I convert into a discussion because, this is not an issue.

**Reply by mzst123** [reply]
Hello, I have the same problem. How do i get a sensor wich is negative if production exceeds consumption? i only have separate sensorts (import, export and load). How do i combine them to make a new sensor with this requisite?

**Reply by jmcollin78** [reply]
You can create Helpers sensors or template like I does. Something like this:

```
sensor:
    - name: "Total puissance consommée net instantanée (W)"
      unique_id: total_power_consommee_net_w
      unit_of_measurement: "W"
      device_class: power
      state_class: measurement
      state: >
        {% set importe = states('sensor.total_puissance_importee_instantanee_w') | float(default=-1.0) %}
        {% set exporte = states('sensor.total_puissance_exportee_instantanee_w') | float(default=-1.0) %}
        {% if importe < 0 or exporte < 0  %}{{none}}{% else %}
          {% set delta = (importe - exporte) | round(2) %}
             {{ delta }}
        {% endif %}
```
